purchase_request_procurement/models/procurement.py

```diff
@@ -84,12 +84,13 @@ def propagate_cancels(self):
             # Remove the purchase request lines, if the request is not draft
             # or reject
             for line in request_lines:
-                if line.request_id.state not in ('draft', 'reject'):
-                    raise ValidationError(_('Can not cancel this procurement '
-                                            'as the related purchase request '
-                                            'is in progress confirmed already.'
-                                            ' Please cancel the purchase'
-                                            ' request first.'))
+                if (line.request_id.state not in ('draft', 'reject') and not
+                        self._context.get('from_purchase_request')):
```

Comment:
If you cancel a procurement which is linked to a draft purchase request, an error will be raised. Are you sure this is the behavior we want?